idle_inhibit: Session lock and unmapped surface fixes

kennylevinsen opened the pull request:
We were not correctly handling idle inhibitor visibility in case of session locks, nor in case of unmapped surfaces.

This tries to fix both. Not yet tested.

WhyNotHugo commented:
Does not fix https://github.com/swaywm/sway/issues/8080

Idle is still inhibited if a layer shell is never committed and never made visible.

kennylevinsen commented:
> Idle is still inhibited if a layer shell is never committed and never made visible.

I have added an explicit map check to layer and lock surfaces, and added a test for whether the layer surface output is enabled while we were at it. Could I get you to retest if that did the trick?

WhyNotHugo commented:
Er, I think I mis-tested yesterday.

f1b88377112ce2f2742624b9401dec3cf27ebcff works, but cce747cd4dbb8460286605434ed8469a3d6cb014 also works fine.

WhyNotHugo commented:
I think that cce747cd4dbb8460286605434ed8469a3d6cb014 might be unnecessary?

WhyNotHugo commented:
Needs rebase, won't build with latest wlroots:

```
[176/375] Compiling C object sway/sway.p/input_seatop_resize_tiling.c.o
[177/375] Compiling C object sway/sway.p/input_seatop_resize_floating.c.o
[178/375] Compiling C object sway/sway.p/input_seatop_move_tiling.c.o
[179/375] Compiling C object sway/sway.p/input_seatop_move_floating.c.o
ninja: job failed: cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Isubprojects/wlroots/include -I../subprojects/wlroots/include -Iprotocols -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/elogind -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -DWLR_USE_UNSTABLE -D_POSIX_C_SOURCE=200809L -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wno-format-zero-length -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.10-dev-f1b883771 (" __DATE__ ", branch '"'"'pr8175'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/input_tablet.c.o -MF sway/sway.p/input_tablet.c.o.d -o sway/sway.p/input_tablet.c.o -c ../sway/input/tablet.c
In file included from ../sway/input/tablet.c:3:
../subprojects/wlroots/include/wlr/types/wlr_tablet_v2.h:15:10: fatal error: tablet-v2-protocol.h: No such file or directory
   15 | #include "tablet-v2-protocol.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: subcommand failed
```

I tested f1b88377112ce2f2742624b9401dec3cf27ebcff with an older wlroots and can confirm that this fixes https://github.com/swaywm/sway/issues/8080

If I run my test code (which creates a create a layer-shell surface, attaches an idle inhibitor to it, but never maps it). The idle timeouts trigger as expected.

`wayidle -t 3` times out after three seconds.

emersion reviewed (approved):
LGTM, thanks!
